tests/filecheck/dialects/riscv/riscv_register_allocation_linear_scan.mlir

```diff
@@ -0,0 +1,71 @@
+// RUN: xdsl-opt -p riscv-allocate-registers{allocation_strategy=LinearScan} %s --print-op-generic | filecheck %s
+
+"builtin.module"() ({
+  %0 = "riscv.li"() {"immediate" = 6 : i32} : () -> !riscv.reg<>
```

Comment:
I feel this test case would very much benefit from custom syntax (https://github.com/xdslproject/xdsl/issues/1067).

Also, I am not fully sure what this test case is testing? It seems to contain a sequence of unused constants that are all assigned to the register. Does this cover any interesting corner case of the linar scan register allocator?

Would it make sense to add an option to run the register allocator with a limited register set, e.g., just 3, and use this to write smaller test cases?

Reply:
The test suite is yet to be improved indeed. The current file check was mainly here to ensure that there was not regression over the `block_naive` strategy (so not replacing already allocated registers).
The API for limiting the number of available registers is already there but not exposed yet as an attribute, but it will be used for future tests.

Reply:
We also have support for pass arguments (added by @AntonLydike) that allow your filecheck test cases to set such local variables.

Reply:
@adutilleul, what do you think about custom syntax? I feel this could be a separate patch that will make this patch a lot nicer to review.